File: py-rsmdu/examples/lcz_from_url.py
"""
Example: Download LCZ (Local Climate Zone) data from URL using rsmdu

This example demonstrates how to:
1. Create an Lcz instance
2. Set a bounding box
3. Load LCZ data from zip URL
4. Get GeoJSON data
5. Save to GPKG file
"""

import rsmdu


def main():
    print("🌡️  Loading LCZ from URL...")

    # Create Lcz instance
    lcz = rsmdu.geometric.Lcz(output_path="./output")

    # Set bounding box (La Rochelle area, France)
    # Format: min_x, min_y, max_x, max_y (WGS84, EPSG:4326)
    lcz.set_bbox(-1.152704, 46.181627, -1.139893, 46.18699)

    # Set CRS (optional, defaults to EPSG:2154)
    lcz.set_crs(2154)

    geo = lcz.geo_core
    print(f"📦 Bounding box set")
    print(f"🗺️  CRS: {geo.epsg}")
    print(f"📁 Output path: {geo.output_path}")

    # Display LCZ color table
    print("\n🎨 LCZ Color Table:")
    table_color = lcz.get_table_color()
    for code in sorted(table_color.keys()):
        info = table_color[code]
        print(f"  LCZ {code}: {info['name']} ({info['color']})")

    # Run LCZ processing: loads from zip URL, filters by bbox
    print("\n⏳ Loading LCZ from URL...")
    print("  Note: Full implementation requires GDAL shapefile reading")
    lcz = lcz.run()

    # Get GeoJSON (equivalent to to_gdf() in Python)
    print("📊 Getting GeoJSON data...")
    geojson = lcz.get_geojson()

    if geojson and "features" in geojson:
        num_features = len(geojson["features"])
        print(f"✅ Loaded {num_features} LCZ zones")

        # Save to GPKG
        print("💾 Saving to GPKG...")
        lcz.to_gpkg(name="lcz")
    else:
        print("⚠️  LCZ processing not yet fully implemented")
        print("   Full implementation requires:")
        print("   - GDAL shapefile reading from zip URLs")
        print("   - Spatial overlay operations")
        print("   - CRS reprojection")
        print("   For now, LCZ structure is ready but data loading is pending.")

    print(f"✅ LCZ processing complete!")
    print(f"📁 Output path: {lcz.get_output_path()}")

    return lcz


if __name__ == "__main__":
    lcz = main()

File: rsmdu/examples/lcz_from_url.rs
use anyhow::Result;
use rsmdu::geometric::lcz::Lcz;

/// Example: Loading LCZ (Local Climate Zone) data from URL
/// Following Python example from pymdu.geometric.Lcz
fn main() -> Result<()> {
    println!("=== Example: Loading LCZ from URL ===\n");

    // Create Lcz instance
    // Python: lcz = Lcz()
    let mut lcz = Lcz::new(None, Some("./output".to_string()), None)?;

    // Set bounding box (La Rochelle, France)
    // Python: lcz.bbox = [-1.152704, 46.181627, -1.139893, 46.18699]
    lcz.set_bbox(-1.152704, 46.181627, -1.139893, 46.18699);

    println!("Bounding box set:");
    println!("  - Longitude: -1.152704 to -1.139893");
    println!("  - Latitude: 46.181627 to 46.18699");
    println!("  - Area: La Rochelle, France");
    println!("  - Format: WGS84 (EPSG:4326)\n");

    // Display LCZ color table
    println!("LCZ color table:");
    for (code, (name, color)) in &lcz.table_color {
        println!("  LCZ {}: {} ({})", code, name, color);
    }
    println!();

    // Run LCZ processing
    // Python: lcz = lcz.run()
    println!("Downloading and processing LCZ from URL...");
    println!("  Note: Full implementation requires GDAL shapefile reading and spatial overlay");
    let lcz_result = lcz.run(None)?;

    println!("\n✅ Processing complete!");
    println!("  - TODO: Implement full shapefile reading and overlay operations");

    Ok(())
}
